libbeat/processors/rate_limit/rate_limit.go

```diff
@@ -0,0 +1,133 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package rate_limit
+
+import (
+	"fmt"
+	"sort"
+
+	"github.com/jonboulle/clockwork"
+	"github.com/mitchellh/hashstructure"
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/v7/libbeat/beat"
+	"github.com/elastic/beats/v7/libbeat/common"
+	"github.com/elastic/beats/v7/libbeat/logp"
+	"github.com/elastic/beats/v7/libbeat/processors"
+	"github.com/elastic/beats/v7/libbeat/processors/rate_limit/algorithm"
+	jsprocessor "github.com/elastic/beats/v7/libbeat/processors/script/javascript/module/processor"
+)
+
+func init() {
+	processors.RegisterPlugin("rate_limit", New)
+	jsprocessor.RegisterPlugin("RateLimit", New)
```

Comment:
Thinking twice about this, I don't think this processor should be available in the script processor. The script processor has a pool of sessions that can be started and destroyed as needed. Each session has its own instances of the processors used. This works well for stateless processors, but not for processors with some state as this one (related discussion about this in https://github.com/elastic/beats/pull/16349).
For this case this would mean that each session would have its own rate limiter, each one with their own buckets, that could be replaced at any moment, this would make it of little use.
```suggestion
```
The `rate_limit` processor could be used in any case before or after the `script` processor to rate limit events related to the script.

Reply:
Removed in ac712047fd3a0e80167e44b766b328532e06c5a8.